src/rgw/rgw_sal_motr.cc

```diff
@@ -1561,6 +1551,16 @@ int MotrObject::MotrDeleteOp::delete_obj(const DoutPrefixProvider* dpp, optional
     return rc;
   }
 
+  // Finally, delete the object's entry from the bucket index.
```

Comment:
Only after deleting the old object (if any exists), delete entry from the bucket index

Reply:
For zero size object, we are not going to delete the motr object and we return at line number 1542. In that case we will not able to delete the metadata for zero size object. We need to remove the  `return` statement also make sure we don't try to delete the motr object for zero size object.

Reply:
done. Changes done to make sure S3 metadata for zero sized old object is deleted